overlord/snapstate/snapstate_test.go

```diff
@@ -101,7 +101,7 @@ func (s *snapmgrBaseTest) settle(c *C) {
 	s.state.Unlock()
 	defer s.state.Lock()
 
-	err := s.o.Settle(testutil.HostScaledTimeout(5 * time.Second))
+	err := s.o.Settle(testutil.HostScaledTimeout(15 * time.Second))
```

Comment:
Excellent questions. Their identical numbers were actually a coincidence. I was experimenting with how much time focal (because this is only failing on the google focal machine) required to pass its tests. When I increased the settle time, rather than die due to not converging, the tests would reach the overall unit test timeout. So I keep increasing each of those two values by five until the tests were able to pass. It looks like I overshot the settle though. I just dialed that back by 5 seconds and focal is still happy.

Reply:
Hmm I'm torn about increasing the unit test timeout universally. On the one hand, unit tests ought to take the same amount of time on all systems (else they're not very "unit") (also, why is it taking focal 3x as long??), but on the other hand, if there's a deadlock in a unit test, the runner will sit spinning as long as the timeout lets it, so it could just be burning money. Deadlocks shouldn't be happening though, and local testing before pushing would find them, so I think I'm in favor of universally increasing timeouts, especially since this is for running unit tests inside spread, where runners may be glacial.

Reply:
I think the issue has to do with the machine itself (Perhaps network connectivity? Perhaps the CPU is overloaded?). I tried using various focal VMs (both lxd and qemu) and had no issues whatsoever; the unit tests completed in roughly the same amount of time as it takes on my native system.

It would be really nice to not mix the particularities of our CI environment with the codified correct behavior of snapd. That does muddy the water and makes it hard to understand why tests do what they do. It would be awesome to extract these kind of things into configuration files or something that are specific to the backend.